eksperymenty_programistyczne_na_edoku4.rst

```diff
@@ -0,0 +1,91 @@
+
+Po stronie zewnętrznej aplikacji należało opanować API (format danych, sposób autoryzacji, adresy endpointów). Zewnętrzna aplikacja została napisana w pythonie (narzędzie wywoływane z linii poleceń oraz serwer HTTP serwujący prosty interfejs graficzny). Użycie REST API eDoka4 nie sprawiło problemów. EDok4 nie posiada dokumentacji swojego API poza kodem.
+
+Podczas tworzenia oprogramowania napotkano problemy:
+
+* mieszanie polskiego i angielskiego nazewnictwa w kodzie - utrudnia to pracę programistom (np. trudno zapamiętać czy pole w rekordzie ma nazwę ``client`` czy ``klient``)
+* skonfigurowanie wyszukiwania dokładnym dopasowaniem w silniku wyszukiwania (elasticsearch) było uciążliwe i w rezultacie nie zostało wykonane. Takie wyszukiwanie jest bardzo potrzebne w przypadku szukania po adresie e-mail lub po identyfikatorze klienta.
+
+Wnioski
+~~~~~~~
+
+System eDok4 ma możliwości rozbudowy. Można zintegrować z nim zewnętrzną aplikację, która korzysta z danych w nim zgromadzonych. Możliwość dostępu do danych nie jest ograniczona przez zakres API - można go poszerzyć dodając końcówki i filtry w kodzie aplikacji.
+
+Drugi eksperyment programistyczny
+---------------------------------
+
+Cel technologiczny
+~~~~~~~~~~~~~~~~~~
+
+Lepiej zrozumieć, jak zewnętrzna firma może rozwijać system eDOK. Między innymi:
+
+* dodawać pola na metadane
+* modyfikować widoki interfejsu WWW
+* dodawać rodzaje raportów
+
+
+Opis eksperymentu
+~~~~~~~~~~~~~~~~~
+
+Eksperyment zakłada dobudowanie narzędzia obliczającego stopień zrozumiałości pism wytwarzanych przez urzędników i adresowanych do obywateli, a także uwzględnienie obliczonej wartości w narzędziach do raportowania — pozwalając na określenie zrozumiałości m.in. ze względu na komórkę organizacyjną, klasyfikację JRWA i osobę urzędnika. Dodatkową funkcjonalnością będzie wyświetlanie stopnia zrozumiałości pisma przed jego dodaniem do akt sprawy, aby urzędnik je tworzący miał możliwość dokonania ewentualnej korekty.
+
+Aplikacje oceniające zrozumiałość tekstu już istnieją, np. `Jasnopis
+<http://jasnopis.pl/>`_ (którego kod nie jest publicznie dostępny). W eksperymencie nie będzie jednak przeprowadzona integracja z rzeczywistą aplikacją, tylko zastosowany kikut takowej (w formie funkcji zwracającej bądź to losową wartość, bądź uproszczoną metrykę opartą o dane takie, jak liczba poszczególnych znaków interpunkcyjnych, liczba słów i liczba znaków).
+
+Realizacja eksperymentu na eDoku4
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Podczas realizacji eksperymentu powstał kod, który wprowadza
+
+* zmiany w eDoku
+
+  * nowe pole "clarity score" w metadanych dokumentu
+  * wykonywanie zapytań (REST API) do serwera zewnętrznego oceniającego zrozumiałość
+  * wyświetlanie ocen w interfejsie użytkownika
+  * generowanie rankingów zrozumiałości (jako raportów)
+
+* prosty serwer oceniający zrozumiałość napisany w pythonie ("zatyczka" - faktyczna ocena pism nie była celem)
```

Comment:
Mam poważne wątpliwości czy konieczne jest wprowadzenie pojęcia "zatyczka". Wcześniej był "kikut". Proponuje albo użyć pojęcia "atrapa" (patrz powyższa uwaga), albo całkowicie usunąć nawias. Większa ilość własnych terminów w tekście tym tekst jest trudniejszy w odczytaniu, a to nowe pojęcie w istocie niewiele wnosi do tekstu.